src/io/shared_fd.rs

```diff
@@ -156,3 +198,58 @@ impl Drop for Inner {
         }
     }
 }
+
+// Enum and traits copied from the io-uring crate.
+
+/// A file descriptor that has not been registered with io_uring.
+#[derive(Debug, Clone, Copy)]
+pub struct Raw(pub RawFd); // Note: io-uring names this Fd
+
+/// A file descriptor that has been registered with io_uring using
+/// [`Submitter::register_files`](crate::Submitter::register_files) or [`Submitter::register_files_sparse`](crate::Submitter::register_files_sparse).
+/// This can reduce overhead compared to using [`Fd`] in some cases.
+#[derive(Debug, Clone, Copy)]
+pub struct Fixed(pub u32); // TODO consider renaming to Direct (but uring docs use both Fixed descriptor and Direct descriptor)
```

Comment:
Do we want to allow users to create their own fixed fds like this?

Reply:
That's a good question I don't know the answer to. On one level, it's just a different kind of number, or different type of number. Like a user can create their own file descriptor values if they happen to want to.

On the other hand, if the user's don't need it, not providing certainly reduces their confusion when confronted so much new in an API.

There could be some APIs where the user decides how to partition the fixed table space - we can't be the arbiters for all uses. So I guess that's why I'm inclined to expose the type, for now. I still prefer thinking of this crate as a means to marry tokio with urings, and let the user decide what they will do with it.

I also think its okay to expose things on a trial bases reserving the right to change things in the next breaking version. Otherwise we might not have enough to reach a critical mass.

But if you think we can build what we need without this being public and we want to ask for someone to say they need to do their work, we can revisit the public nature of that type.

Reply:
This is gone.